src/components/ProgressItem.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from "react"
+import { useContext } from "react"
+import { RegexContext } from "../store/regex-context"
+import classes from "./ProgressItem.module.css"
+
+import Card from "../UI/Card"
+
+const ProgressItem: React.FC<{ word: string; wordIndex: number }> = React.memo(
+  (props) => {
+    const [hover, setHover] = useState<boolean>(false)
+    const regExContext = useContext(RegexContext)
+    const { guessedRegExArray } = regExContext
+
+    const cardHoverHandler = (
+      e: React.MouseEvent<HTMLHeadingElement, MouseEvent>
+    ) => {
+      setHover(true)
+    }
+    const cardLeaveHandler = (
+      e: React.MouseEvent<HTMLHeadingElement, MouseEvent>
+    ) => {
+      setHover(false)
+    }
+
+    let guessedRegExArrayTransformed: string[] = []
+
+    for (let value of guessedRegExArray) {
+      guessedRegExArrayTransformed = [
+        ...guessedRegExArrayTransformed,
+        value,
+        value,
+        value,
+      ]
+    }
+
+    return (
+      <Card>
+        <h3 onMouseEnter={cardHoverHandler} onMouseLeave={cardLeaveHandler} className={classes.wordName}>
+          {hover
+            ? guessedRegExArrayTransformed.filter(
```

Comment:
probably, it's better to move this function away from render function

Reply:
Sure, that makes sense! Will redo!